src/main/java/org/folio/orders/utils/validators/CompositePoLineValidationUtil.java

```diff
@@ -0,0 +1,221 @@
+      return validatePoLineWithElectronicFormat(compPOL);
+    } else if (orderFormat == CompositePoLine.OrderFormat.PHYSICAL_RESOURCE) {
+      return validatePoLineWithPhysicalFormat(compPOL);
+    } else if (orderFormat == CompositePoLine.OrderFormat.OTHER) {
+      return validatePoLineWithOtherFormat(compPOL);
+    }
+
+    return Collections.emptyList();
+  }
+
+  private static List<Error> validatePoLineWithMixedFormat(CompositePoLine compPOL) {
+
+    List<ErrorCodes> errors = new ArrayList<>();
+    // The quantity of the physical and electronic resources in the cost must be specified
+    if (HelperUtils.getPhysicalCostQuantity(compPOL) == 0) {
+      errors.add(ErrorCodes.ZERO_COST_PHYSICAL_QTY);
+    }
+    if (HelperUtils.getElectronicCostQuantity(compPOL) == 0) {
+      errors.add(ErrorCodes.ZERO_COST_ELECTRONIC_QTY);
+    }
+
+    errors.addAll(HelperUtils.validateLocations(compPOL));
+    errors.addAll(validateCostPrices(compPOL));
+
+    return convertErrorCodesToErrors(compPOL, errors);
+  }
+
+  public static Optional<Error> checkMaterialAvailability(CompositePoLine compPOL) {
+    boolean isMissing = false;
+    if (compPOL.getOrderFormat()
+      .equals(ELECTRONIC_RESOURCE)
+        || compPOL.getOrderFormat()
+          .equals(P_E_MIX)) {
+      isMissing = compPOL.getEresource()
+        .getCreateInventory() == (Eresource.CreateInventory.INSTANCE_HOLDING_ITEM) && isEmpty(
+            compPOL.getEresource()
+              .getMaterialType());
+    }
+
+    if (!compPOL.getOrderFormat()
+      .equals(ELECTRONIC_RESOURCE)) {
+      isMissing = isMissing || (compPOL.getPhysical()
+        .getCreateInventory() == Physical.CreateInventory.INSTANCE_HOLDING_ITEM && isEmpty(
+            compPOL.getPhysical()
+              .getMaterialType()));
+    }
+
+    if (isMissing) {
+      return Optional.of(convertErrorCodesToError(compPOL, ErrorCodes.MISSING_MATERIAL_TYPE));
+    }
+    return Optional.empty();
+  }
+
+  public static List<Error> checkMaterialsAvailability(List<CompositePoLine> poLines) {
+    if (Objects.nonNull(poLines)) {
+      return poLines.stream()
+        .map(CompositePoLineValidationUtil::checkMaterialAvailability)
+        .filter(Optional::isPresent)
+        .map(Optional::get)
+        .collect(toList());
+    }
+    return Collections.emptyList();
+  }
+
+  private static List<Error> validatePoLineWithPhysicalFormat(CompositePoLine compPOL) {
+    List<ErrorCodes> errors = new ArrayList<>();
+
+    // The quantity of the physical resources in the cost must be specified
+    if (HelperUtils.getPhysicalCostQuantity(compPOL) == 0) {
+      errors.add(ErrorCodes.ZERO_COST_PHYSICAL_QTY);
+    }
+    // The quantity of the electronic resources in the cost must not be specified
+    if (HelperUtils.getElectronicCostQuantity(compPOL) > 0) {
+      errors.add(ErrorCodes.NON_ZERO_COST_ELECTRONIC_QTY);
+    }
+
+    errors.addAll(HelperUtils.validateLocations(compPOL));
```

Comment:
`HelperUtils.validateLocations` also related to PoLine verification and called only from `CompositePoLineValidationUtil`, so I think it should be moved to `CompositePoLineValidationUtil`

Reply:
Done